controller/src/controller/storage/kubernetes/fsadmin.py

```diff
@@ -0,0 +1,152 @@
+"""Kubernetes storage layer for filesystem administration pod."""
+
+# As with the fsadmin service, this is very similar to the fileserver, but
+# lacks a username.
+
+from __future__ import annotations
+
+from kubernetes_asyncio.client import ApiClient
+from structlog.stdlib import BoundLogger
+
+from ...config import FSAdminConfig
+from ...models.domain.fsadmin import FSAdminObjects
+from ...models.domain.kubernetes import PodPhase
+from ...timeout import Timeout
+from ..metadata import MetadataStorage
+from .deleter import PersistentVolumeClaimStorage
+from .pod import PodStorage
+
+__all__ = ["FSAdminStorage"]
+
+
+class FSAdminStorage:
+    """Kubernetes storage layer for a filesystem admin pod.
+
+    Parameters
+    ----------
+    config
+        Configuration for fsadmin environment.
+    metadata_storage
+        Holds namespace information.
+    api_client
+        Kubernetes API client.
+    logger
+        Logger to use.
+
+    Notes
+    -----
+    This class isn't strictly necessary; instead, the fsadmin service
+    could call the storage layers for individual Kubernetes objects
+    directly. Even though there are not many objects, having a wrapper layer
+    might be easier to follow.
+    """
+
+    def __init__(
+        self,
+        config: FSAdminConfig,
+        metadata_storage: MetadataStorage,
+        api_client: ApiClient,
+        logger: BoundLogger,
```

Comment:
Nit: Over three parameters, so this should force keyword arguments with `*,` before the arguments.